lib/npa.js

```diff
@@ -87,6 +88,23 @@ function resolve (name, spec, where, arg) {
   }
 }
 
+function toPurl (name, spec) {
```

Comment:
This should accept a package arg. It's the job of this module to then break it up into a name and rawSpec.  You can then check the type of the result, if it's `version` you know it already passed `semver.version`.